### DISCUSSION TITLE
load BTF: invalid argument: [201]

### DISCUSSION THREAD

**Discussion by Ephemeraler** [cilium/ebpf#811] [Q&A]
Please，What does this error code mean。

```
// kprobe.c

SEC("kprobe/vfs_read")
int BPF_KPROBE(vfs_read, struct file *filp, char *buf, size_t count, loff_t *pos)
{
    struct inode* inode;
    bpf_probe_read_kernel(&inode, sizeof(inode), &filp>f_inode);
    bpf_probe_read_kernel(&sample.filesize_, sizeof(sample.filesize_), &inode->i_size);


    /* get filesystem. */
    struct super_block *sb;
    bpf_probe_read_kernel(&sb, sizeof(sb), &inode->i_sb);
    bpf_probe_read_kernel(sample.filesystem_name_, sizeof(sample.filesystem_name_), sb->s_id);
}
```

the error happen at sb->s_id, but I use libbpf, this code could run.

error message:
load BTF: invalid argument: [199] STRUCT sb_writers size=320 vlen=3: wait_unfrozen type_id=103 bits_offset=64 Invalid member

**Comment by lmb** [comment]
Hi! Please answer the following questions:
- What kernel version is your `vmlinux.h` from?
- What kernel version are you trying to load this on?
- Are you on `master` of `github.com/cilium/ebpf`?
- Can you put a reproducer somewhere?
